src/event_logger.cpp

```diff
@@ -110,4 +110,76 @@ void EventLogger::createAsyncPelWithI2cBusCallout(
     (void)i_userData1;
     (void)i_userData2;
 }
+
+void EventLogger::createAsyncPel(
+    const types::ErrorType i_errorType, const types::SeverityType i_severity,
+    const std::string& i_fileName, const std::string& i_funcName,
+    const std::string& i_internalRc,
+    const std::optional<std::pair<std::string, std::string>> i_userData1,
+    const std::optional<std::pair<std::string, std::string>> i_userData2)
+{
+    try
+    {
+        if (m_errorMsgMap.find(i_errorType) == m_errorMsgMap.end())
+        {
+            logging::logMessage(
+                "PEL error type not found in the error message map.");
+            return;
+        }
+
+        const std::string& l_message = m_errorMsgMap.at(i_errorType);
+
+        const std::string& l_severity =
+            (m_severityMap.find(i_severity) != m_severityMap.end()
+                 ? m_severityMap.at(i_severity)
+                 : m_severityMap.at(types::SeverityType::Error));
+
+        std::string l_description("Description field missing in user data");
+
+        if (i_userData1 && ((*i_userData1).first.compare("DESCRIPTION") ==
+                            constants::STR_CMP_SUCCESS))
+        {
+            l_description = (*i_userData1).second;
+        }
+
+        sd_bus* l_sdBus = nullptr;
+        sd_bus_default(&l_sdBus);
+
+        auto l_rc = -1;
+        if (!i_userData2)
```

Comment:
why we need both userData 1 and userData2? 

userData1 is an std::pair which can hold n number of pairs. so can we stick with one userData only?

Reply:
userData1 and userData2 are kept if anything comes in the future, caller can use these to send more additional data.